### ISSUE TITLE
Custom trend disappears from handlesummary output to console.

### ISSUE THREAD

**Issue by MattDodsonEnglish** [grafana/k6#3042]
### Brief summary

In this script, one custom trend, `wsw_time`, disappears from the console output when I use handleSummary .
Strangely, I think it didn't happen _every_ time, either. Sorry I can't confirm that it's only occasionally reproducible, but I can attach a screenshot of a time it didn't appear.

```
import http from "k6/http";
import { textSummary } from "https://jslib.k6.io/k6-summary/0.0.2/index.js";
import { Trend } from "k6/metrics";

const myTrend = new Trend("wsw_time");
const theirTrend = new Trend("their_time");

export default function () {
  const getRes = http.get("http://httpbin.test.k6.io/");
  theirTrend.add(getRes.timings.duration);
  const resWSW = http.get("https://wellshapedwords.com/");
  myTrend.add(resWSW.timings.duration);
}

export function handleSummary(data) {
  return {
    stdout: textSummary(data, { indent: "→", enableColors: true }),
  };
}
```

Not that the other custom metric, `their_time` shows up fine in both cases.

![image](https://user-images.githubusercontent.com/47385188/234942560-bb846e24-20bf-40ef-b748-34d9fec9a720.png)

### k6 version

k6 v0.44.0 ((devel), go1.20.3, linux/amd64)

### OS

Arch linux, installed through AUR

### Steps to reproduce the problem

Run script attached

### Expected behaviour

`wsw_time` summary appears in handleSummary terminal output.

### Actual behaviour

`wsw_time` doesn't appear.

**Comment by mstoykov** [comment]
This is a bug ... of sorts. 

The in k6 implementation does by default add a bunch of new lines. And I feel like that should just not be needed. As without them parts of the end summary got eaten up by other parts that are written over.

https://github.com/grafana/k6/blob/9fa50b2d1f259cdccff5cc7bc18a236d31c345ac/js/summary-wrapper.js#L66

I don't know if this was done due to some kind of deficiency that we intend on fixing or what.

cc @na-- any memories as you were the one that implemented this originally 

As current workaround - copy the linked code.

**Comment by na--** [comment]
> I don't know if this was done due to some kind of deficiency that we intend on fixing or what.

No, the newlines definitely were not to avoid overwriting by the progress bars, they were to make the JS summary output identical to the old k6 Go code that produced the same summary.

The problem probably occurs because we use terminal control codes to control the terminal "cursor" (caret?) and essentially print the progressbars in the same place. I might have messed up something in https://github.com/grafana/k6/pull/2815 that caused the last progressbar printing to overwrite the last line of the end-of-test summary :thinking: Or this bug might have existed before even that.

In either case, we should probably finish with the progress bars _before_ we handle the end-of-test summary, which should resolve this bug too :thinking:
